### PR TITLE
Expose deployment artifacts from deployments package rather than ABI + bytecode

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

Rather than pulling out the ABI and bytecode for each contract and saving them to the `bytecode`/`abi` folders, we now have a single `artifact` folder per deployment task. This is then populated from the build-info file in the exact same way that hardhat generates its own artifact files so users of the npm package can use these as if they had compiled the contracts locally.

`getBalancerContractABI` is not deprecated as this is a useful helper for users who only want to be able to interact with existing contracts but not deploy them. For anyone deploying their own contracts then it's expected that they use `getBalancerContractArtifact`. `getBalancerContractBytecode` is deprecated.

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests are included for all code paths
- [x] The base branch is either `master`, or there's a description of how to merge

### CODE REVIEW

**Comment by TomAFrench** [comment]
Adding changelog stuff now.

**Comment by TomAFrench** [comment]
There's also a load of really mangled english in the jsdoc here. Will fix in a separate PR.

File: pkg/deployments/src/artifact.ts [hunk 1]
```diff
@@ -44,69 +43,88 @@ export function checkArtifact(task: Task): void {
       const fileName = path.parse(buildInfoFileName).name;
       const contractName = fileName;
 
-      const expectedArtifact = extractContractArtifact(task, fileName, contractName);
-      const { abi, bytecode } = readContractABIAndBytecode(task, contractName);
+      const expectedArtifact = task.artifact(contractName, fileName);
+      const artifact = readContractArtifact(task, contractName);
```
Comment (nventuro) [inline pkg/deployments/src/artifact.ts:47]:
```suggestion
      const actualArtifact = readContractArtifact(task, contractName);
```

File: pkg/deployments/src/artifact.ts [hunk 2]
```diff
@@ -44,69 +43,88 @@ export function checkArtifact(task: Task): void {
...
 
-      const bytecodeMatch = bytecode === expectedArtifact.evm.bytecode.object;
-      const abiMatch = JSON.stringify(abi) === JSON.stringify(expectedArtifact.abi);
-      if (bytecodeMatch && abiMatch) {
-        logger.success(`Verified ABI and bytecode integrity of contract '${contractName}' of task '${task.id}'`);
+      const artifactMatch = JSON.stringify(artifact) === JSON.stringify(expectedArtifact);
```
Comment (nventuro) [inline pkg/deployments/src/artifact.ts:49]:
```suggestion
      const artifactMatch = JSON.stringify(actualArtifact) === JSON.stringify(expectedArtifact);
```

**Review by nventuro — APPROVED** [review]

File: pkg/deployments/index.ts [hunk 3]
```diff
@@ -18,28 +19,40 @@ export async function getBalancerContract(task: string, contract: string, networ
...
+  return require(getBalancerContractArtifactPath(task, contract));
 }
 
 /**
+ * @dev Returns the ABI for a contract from a specific task
+ * @param task ID of the task to look the ABI of the required contract
+ * @param contract Name of the contract to be fetched.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getBalancerContractABI(task: string, contract: string): any[] {
```
[the remark this answers is not in the document]
Comment (TomAFrench) [reply pkg/deployments/index.ts:42]: Ack, managed to delete this function and didn't notice this when I re-added it.

**Comment by TomAFrench** [comment]
@nventuro Just realised that this PR means that we need to go update all the links in each deployment :sadge:

**Comment by nventuro** [comment]
'we'

**Pull request merged**